internal/pkg/service/buffer/api/service/service.go

```diff
@@ -181,10 +184,84 @@ func (*service) DeleteExport(dependencies.ForProjectRequest, *buffer.DeleteExpor
 	return nil, &NotImplementedError{}
 }
 
-func (*service) Import(dependencies.ForPublicRequest, *buffer.ImportPayload, io.ReadCloser) (err error) {
+func (*service) Import(d dependencies.ForPublicRequest, payload *buffer.ImportPayload, reader io.ReadCloser) (err error) {
+	ctx, store := d.RequestCtx(), d.ConfigStore()
+	receiver, err := store.GetReceiver(ctx, payload.ProjectID, payload.ReceiverID)
+	if err != nil {
+		return err
+	}
+	if receiver.Secret != payload.Secret {
+		return &GenericError{
+			StatusCode: 404,
+			Name:       "buffer.receiverNotFound",
+			Message:    fmt.Sprintf(`Receiver "%s" with given secret not found.`, payload.ReceiverID),
+		}
+	}
+
+	data, err := readCSVFromRequest(payload.ContentType, reader)
+	if err != nil {
+		return err
+	}
+
+	exports, err := store.ListExports(d.RequestCtx(), payload.ProjectID, payload.ReceiverID)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range exports {
+		mapping, err := store.GetCurrentMapping(ctx, payload.ProjectID, payload.ReceiverID, e.ID)
+		if err != nil {
+			return err
+		}
+		for _, c := range mapping.Columns {
+			//TODO
+		}
```

Comment:
@michaljurecko Takže podle [issue](https://keboola.atlassian.net/browse/BA-61?focusedCommentId=49083) iteruju `mapping.Columns` a jdu generovat csv. Nějak takto:

![CleanShot 2022-11-21 at 22 10 43@2x](https://user-images.githubusercontent.com/215660/203158910-2bfa4af3-9542-40b1-85f7-596ec9a894fb.jpg)

Za použití tý metody `CsvValue()` co jsi psal tady: https://github.com/keboola/keboola-as-code/pull/997#discussion_r1028014471, jo? A co je ten `ImportContext` v parametru, to asi není `context.Context`? To si mám sestavit `map[string]string` z dat který budu potřebovat a potom to vybrat ve stylu:

```go
func (Body) CsvValue(ImportContext map[string]string context.Context) string {
	return ImportContext["body"]
}
```
?

A v [tý issue](https://keboola.atlassian.net/browse/BA-61?focusedCommentId=49083) píšeš že mám podle content-type dekódovat body, ale co s ním potom? V [Design Doc](https://keboola.atlassian.net/wiki/spaces/KB/pages/2793013269/Buffer+API+Design?focusedCommentId=2826076179#Mapping) je napsaný, že sloupec `body` má být "raw HTTP body". Tak možná nějaký sjednocení, ale nevím přesně. A co se týče headers a IP adresy, tak ty dostanu nějak snadno sem do týhle metody, nebo je mám odněkud strčit do `dependencies.ForPublicRequest`?

Reply:
mozeme zajtra prebrat, ... `ImportContext` ma byt vsetko potrebne pre vygenerovanie hodnoty (ip, body ...) ... to body moze byt kludne aj map (asi ordered, zalezi nam na poradi (?))

Reply:
Ještě mi není jasný kde vezmu `fileID` a `sliceID` který potřebuju pro record key.